draft-ietf-jsonpath-base.md

```diff
@@ -902,8 +933,128 @@ size doesn't
+~~~~
+
+#### Semantics
+{: numbered="false" toc="exclude"}
+
+The `descendant-selector` is inspired by ECMAScript for XML (E4X). It selects object members and array elements in a recursive descent process.
+
+### Union Selector
+
+#### Syntax
+
+The union selector is syntactically related to the `index-selector`. It contains multiple, comma separated entries.
+
+~~~~ abnf
+union-selector = "[" ws union-entry 1*(ws "," ws union-enty) ws "]"
 
+union-entry    =  ( quoted-member-name /
+                    element-index      /
+                    slice-index
+                  )
+~~~~
+
+#### Semantics
+
+A union selects any node which is selected by at least one of the union selectors and selects the concatenation of the
+lists (in the order of the selectors) of nodes selected by the union elements.
+Possibly duplicate selected nodes are kept in the node list.
+
+
+### Filter Selector
+
+#### Syntax
+
+The filter selector has the shape `'[?<expr]'`. It works via iterating over container values, i.e. arrays and objects.
+
+~~~~ abnf
+filter-selector    = "[?" boolean-expr "]"
+~~~~
+
+During iteration process each array element or object member is visited and its value &ndash; accessible via symbol `'@'` &ndash; or one of its descendants &ndash; uniquely defined by a relative path &ndash; is tested against a boolean expression `boolean-expr`.
+
+If the result is `true`, the current item, represented by `'@'`, is selected. In case of a `false` result, it is not.
+
+~~~~ abnf
+boolean-expr = *logical-expr
+logical-expr = [neg-op] ["("] comp-expr *[logical-op comp-expr] [")"]
+neg-op       = "!"                                  ; not operator
+logical-op   = "||" / "&&"                          ; logical operator
+comp-expr    = (rel-path-val /
+                calc-val) [(comp-op comparable /  ; comparison
+                            regex-op regex     /  ; RegEx test
+                            in-op container )]    ; containment test
+comp-op      = "==" / "!=" /                        ; comparison ...
+               "<"  / ">"  /                        ; operators
+               "<=" / ">="
+regex-op     = "~="                                 ; RegEx match
+in-op        = " in "                               ; in operator
+comparable   = number / quoted-string /             ; primitive ...
+               true / false / null /                ; values only
+               rel-path-val /                       ; descendant value
+               calc_val /                           ; calculated value
+               json-path                            ; any value
+
+rel-path-val = "@" *(dot-selector / index-selector)
+calc_val     = func "(" [rel-path-val / json-path] ")"
+func         = "index"
+~~~~
+
+Notes:
+
+* Parenthesis can be used with `boolean-expr` for grouping. So filter selection syntax in the original proposal `'[?(<expr)]'` is naturally contained in the current lean syntax `'[?<expr]'` as a special case.
```

Comment:
What is the meaning of `<expr`?
Missing `>`?

Reply:
yes ... simple typo

thanks a lot